File: springServer/src/main/java/com/socialnetwork/user/entities/Comment.java
package com.socialnetwork.user.entities;

import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(name = "comments")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", updatable = false, nullable = false)
    private Long id;
    private String text;
    private Date date;

    @ManyToOne
    @JoinColumn(name = "post_id", insertable = false, updatable = false)
    private Post post;

    @Column(name = "post_id", nullable = false)
    private Long postId;

    @ManyToOne
    @JoinColumn(name = "author_id", referencedColumnName = "id")
    private User author;

    public Comment() {}


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }

    public Long getPostId() {
        return postId;
    }

    public void setPostId(Long postId) {
        this.postId = postId;
    }
}

File: frontApplication/src/app/model/comment.ts
import { Post } from "./post";
import { User } from "./user";

export class Comment {
  constructor(
    private author: User,
    private text: string,
    private date: Date,
    private post: Post
  ) { }

  public get authorName(): string {
    return this.author.name;
  }

  public get Text(): string {
    return this.text;
  }

  public get Date(): Date {
    return this.date;
  }

  public get Post(): Post {
    return this.post;
  }

  public get PostId(): number {
    return this.Post.id;
  }
}